### ISSUE TITLE
Marvel Snap Banned Me

### ISSUE THREAD

**Issue by HolidayPhenom** [AdriaGual/marvel-snap-bot#84]
![5428E70B-F23B-4FDF-B740-891CC983A789](https://user-images.githubusercontent.com/104886313/219388485-7bfae555-16b5-4df0-8676-6c76e688f4d9.png)
![D5DD42E5-AC79-466B-AB35-3EDD5093EA70](https://user-images.githubusercontent.com/104886313/219388532-57aede40-423a-4abc-a2aa-42ec8f61601c.png)
![EF780437-1384-4B20-9149-96C2221AD6D6](https://user-images.githubusercontent.com/104886313/219388556-6a8df033-44c4-47b0-9424-9242668eaa0b.png)
![E8DD268F-224D-45DC-828A-10996BB54F91](https://user-images.githubusercontent.com/104886313/219388565-19d178bf-78a1-44cb-a9fa-7dcae6ae6120.png)

So I’m almost level 3000, they gave me a soft ban for a day and now I’m fearful of using the bot again. It took them about 4 days to notice. I just wanted to let everyone know the bot is catchable and with that info, maybe we can see what measures to take in precaution of reprehension.

**Comment by paperc07** [comment]
Ok what exactly were you doing? The auto concede? Give us all the details cause I been running mine 24 hours straight but it's playing matches non stop all the way to the end. But let's be honest they could still look into it and see how the fuck did this man play this long almost like we need to write a script where it plays for an hour takes a break for 30 mins or something.

**Comment by HolidayPhenom** [comment]
No auto concede (default features only) and I only ran it for 8 hours or so every night :/ possibly 16 hours if I went to work at the most

**Comment by paperc07** [comment]
hmmm cause it said "currently the bot doesn't support early cancellation, please wait patiently for the system to automatically cancel" are they talking about to early cancel your suspension? cause you weren't auto conceding (cancelling) games?

**Comment by paperc07** [comment]
on a side note I ran it atleast 24 hours and got like 9 seasons chests with it even being optimized

**Comment by ZloyPupsic** [comment]
they banned me too, after that I stopped using bot for 8 hours. only missions :(

**Comment by ZloyPupsic** [comment]
> hmmm cause it said "currently the bot doesn't support early cancellation, please wait patiently for the system to automatically cancel" are they talking about to early cancel your suspension? cause you weren't auto conceding (cancelling) games?

they are telling that they are not able no cancel ban early

**Comment by ZloyPupsic** [comment]
maybe random pause between matches can solve the issue

**Comment by paperc07** [comment]
so now that we know we are going to run into this problem we need to make it more human like I suppose?

**Comment by paperc07** [comment]
I mean what else could we do they have bots like hearth ranger and shit that go forever. I mean the more we humanize it how can they really ban you, unless they detect you're running a script

**Comment by HolidayPhenom** [comment]
> I mean what else could we do they have bots like hearth ranger and shit that go forever. I mean the more we humanize it how can they really ban you, unless they detect you're running a script

Ayeeee 🤣 I use hearth ranger too. My Reno Warlock be laying it down 💀 and honestly, I’ve been using hearth ranger for over a year now with no issues

**Comment by isaiahd1** [comment]
dang i was worried about this and have only been testing this bot since yesterday. what should i do? going forward? i dont want to spend all my time farming games for 1k boosters

**Comment by isaiahd1** [comment]
i wish to have a caht discussion for this. but i was thinking of testing multiple decklists to try and confuse the queue. 

like going forward before each match. if a human were to say swap decks maybe every 2-3 games it would look less suspicious. maybe anyone can share deck lists?

**Comment by isaiahd1** [comment]
![image](https://user-images.githubusercontent.com/47872432/219461183-15d27450-612f-4e37-bd2c-5ed43185bf1a.png)
![HD-Player_TcTl9gPwwf](https://user-images.githubusercontent.com/47872432/219461236-91dd3253-e784-4886-82fd-decc3f4a3f1a.png)
these are the two decks im currently rotating i suppose.

**Comment by paperc07** [comment]
so for that your going to need @AdriaGual to make you some code to where it swaps deck's I also think your going to need the image of the named deck for the bot to be able to find and select it

**Comment by isaiahd1** [comment]
well its interesting in any case that they are beginning to detect the behavior any how,. 
![image](https://user-images.githubusercontent.com/47872432/219464288-f74090cc-699a-4722-937f-86df2a6c5745.png)
im just going to keep trying to find kazoo variants to try and test. in general...

**Comment by paperc07** [comment]
I mean of course they were going to start detecting botting lol

And yes there are other archetypes the bot can play but you have to screenshot all the cards in your deck and we are currently trying to work on building a better logic system

**Comment by AdriaGual** [comment]
The bot was never intended to be used 24/7 haha but well, it was a matter of time if people start using it a lot that Marvel Snap would detect it.

Ok so now, here are a couple ideas you can use to play safer:
- Add delays: You can add random delays around the code, it's easy, just use a time.sleep(N) where N is the number of seconds you want the bot to wait, remember to import the time library when you use it, there are plenty of examples in the code, just add more on random spots.
- Edit the `play_random_cards` method, so instead of going throw the same array always, you can rearrange the arrays but iterating them the same way so instead of:
    - I try to play card 1 on field 1, card 2 on field 1, card 3 on field 1.... card N on field N
    It would do something like:
    - I start to play card 3 on field 2, card 4 on field 2.... card 6 on field 1
You can just play with this order.

I'm busy this days with the job, but you can do it! If I can, will review the code and give tips

**Comment by PeanutJar-Jelly** [comment]
I got suspended this morning as well. I don't run the bot for more than an hour or two and I switch out decks alot.

**Comment by paperc07** [comment]
Yeah there definitely gonna be giving me a suspension then

**Comment by isaiahd1** [comment]
> I got suspended this morning as well. I don't run the bot for more than an hour or two and I switch out decks alot.

Jeezzzzzz ! Me personally still not suspended. although this morning myself. I've been having trouble with BlueStacks freezing after 2-3 games.

**Comment by paperc07** [comment]
the bluestacks freezing error is something your probably gonna have to contact them about I know what I did was go into bluestacks settings and change the phone and I changed it to the galaxy s21+ I also know you need virtualization on in your bios I have it set to 4 cores, high performance, and open gl. I also clicked properties and always made it run as  administrator and changed the compatibility to windows 8

**Comment by whimsicott0907** [comment]
I just got suspended as well. I have stopped using the bot for now.

**Comment by paperc07** [comment]
@whimsicott0907 how long were you using it for? did you add any delays? take any breaks? Can you send images on what they said or it just says you've been suspended? is it only for a day? I wonder how many times until a complete ban

They must purposely be letting me get away with it then I have like almost 5 days pretty much 24/7 I need to chill.

**Comment by paperc07** [comment]
> 

So I can add the delays and I added a print function so I see it's working and delaying, but I can't for the life of me get it to delay in between games, I have tried sleeping for so many seconds, I have tried random s delays between 5-15 secs but no matter what it ends game and just clicks next and hurrys up and clicks play

**Comment by paperc07** [comment]
> The bot was never intended to be used 24/7 haha but well, it was a matter of time if people start using it a lot that Marvel Snap would detect it.
> 
> Ok so now, here are a couple ideas you can use to play safer:
> 
> * Add delays: You can add random delays around the code, it's easy, just use a time.sleep(N) where N is the number of seconds you want the bot to wait, remember to import the time library when you use it, there are plenty of examples in the code, just add more on random spots.
> * Edit the `play_random_cards` method, so instead of going throw the same array always, you can rearrange the arrays but iterating them the same way so instead of:
>   
>   * I try to play card 1 on field 1, card 2 on field 1, card 3 on field 1.... card N on field N
>     It would do something like:
>   * I start to play card 3 on field 2, card 4 on field 2.... card 6 on field 1
>     You can just play with this order.
> 
> I'm busy this days with the job, but you can do it! If I can, will review the code and give tips

So I can add the delays and I added a print function so I see it's working and delaying, but I can't for the life of me get it to delay in between games, I have tried sleeping for so many seconds, I have tried random s delays between 5-15 secs but no matter what it ends game and just clicks next and hurrys up and clicks play

**Comment by isaiahd1** [comment]
> Brace Face Baby#4938

hey bro. added you back. breathethroughdiaphragm is my user name. but i figured that. the reason why blue stacks froze must've have to be something with openCV not handling my desktop well. I have fl.ux installed on my pc to change the background color to be more orange and give off lesser light when it's darker out. when i turned fl.ux off the bot ran seamlessly there after.

**Comment by paperc07** [comment]
lmao that's strange but hey whatever as long as it's working

**Comment by whimsicott0907** [comment]
@paperc07 It's the same image as above. I got soft ban for a day. I have used the bot since Zabu season pretty much 24/7 without problem.

**Comment by paperc07** [comment]
oh damn so maybe it just caught up with you, I wish I knew how they were catching people, so we could try to prevent it.

I mean obviously we've all seen the bot run and its retarted but.... I know we can improve it to run better because let's be honest if second dinner taps in and watches our games which we know they do, because that's the purpose of cheater bots they look at your cards you have in hand and then they usually outplay you and win, so if they see this bot dragging let's say sunspot to position 1 then 2 then 3 you can clearly tell a bot is running the account. 

Now I have messed with priorities and its awesome how those work cause you can basically tell it what cards to play based off of average cp but still the way the bot plays needs to be redesigned but that's tough and would take a few coders that know what they're doing and then you could make some progress and trust me this game will pick up coders and you will eventually see a marvel snap bot like Hearthcrawler shit they might even make one and call it SnapCrawler (I know they wont cause I already emailed them) but you get the point

**Comment by nixtynine** [comment]
Same here. I started botting at rank 58, after around 12 hours my ranked dropped to 32. Got suspended for a day. Made an appeal and I can log in once again. Will avoid using the bot in my main until our heroes here figures a way to make it work as 'human' as possible. lols.

Will use it in my alt though.

**Comment by paperc07** [comment]
yeah thats almost guaranteed a part of it lol, but yeah I encourage everyone to make an alt account and bot it as much as you want

**Comment by isaiahd1** [comment]
> 

nixtynine you were only botting for 12 whole hours. like in total? ive been botting for a near week and haven't got whipped yet.

**Comment by nixtynine** [comment]
> > 
> 
> nixtynine you were only botting for 12 whole hours. like in total? ive been botting for a near week and haven't got whipped yet.

lucky you. you farmed a lot.
BTW what rank and collection level are you in?
ill continue botting but only on my alt.

**Comment by Niassamond** [comment]
just got suspended, damn, i hope its also only one day

**Comment by Coldzilla** [comment]
I've been farming for about two weeks. The method is pretty simple:
1) I create 3-4 decks for pumping boosters
2) I set a timer for 1 hour for each deck
3) I restart the bot every hour and change the deck

The maximum game time through the bot is about 3-5 hours.
So far they haven’t been banned, but I would like some updates and improvements to the bot)

And I have a concern about blocking my account not because of using a bot, but because I play from Russia, where the game seems to be blocked by the publisher in the play store

**Comment by paperc07** [comment]
@Niassamond  they got @isaiahd1 today too

**Comment by Niassamond** [comment]
> @Niassamond they got @isaiahd1 today too

did they told him how much time? i still havent received anything about time.

**Comment by Niassamond** [comment]
i sent an appeal been 3 hours no reply yet..

**Comment by isaiahd1** [comment]
yeah. @Niassamond i saw your query on discord too. I'm on there as well trying to appeal. 
a representative.

in game though, the representative took 8 hours to check in on request. 

it also feels like i was flagged. this morning they closed a crash report i sent in a month ago. i had completely forgot it even happened. the ban felt like targeted in some aspect.

well obviously. but i felt like maybe if i cooled down i would've gotten away with it. but truth be told. i was botting right before my suspension. which seemed to be immediately after i closed the bot. and reloaded the game to check the shop on my phone. boom. it was revealed to me that i was banned when i logged on my phone.

**Comment by Niassamond** [comment]
> yeah. @Niassamond i saw your query on discord too. I'm on there as well trying to appeal. a representative.
> 
> in game though, the representative took 8 hours to check in on request.
> 
> it also feels like i was flagged. yesterday morning they closed a crash report i sent in a month ago. i had completely forgot it even happened. the ban felt like I was targeted in some aspect. fast forward maybe 8 hours of botting yesterday.
> 
> well obviously. but i felt like maybe if i cooled down i would've gotten away with it. but truth be told. i was botting right before my suspension. which seemed to be immediately after i closed the bot. and reloaded the game to check the shop on my phone. boom. it was revealed to me that i was banned when i logged on my phone.

did they at least gave you a margin of time to unsuspend or something?

**Comment by isaiahd1** [comment]
> did they at least gave you a margin of time to unsuspend or something?
this is actually strange though. they're giving me two different appeal windows.

whats strange is that when i opened the game on my emulator the representative spoke to me immediately. and its the same guy that spoke to Holiday Phenom. Jay.

**Comment by isaiahd1** [comment]
![image](https://user-images.githubusercontent.com/47872432/220973214-20e4b287-521a-43c7-9802-2a0edd7a032c.png)
DAMN!

**Comment by Niassamond** [comment]
> ![image](https://user-images.githubusercontent.com/47872432/220973214-20e4b287-521a-43c7-9802-2a0edd7a032c.png) DAMN!
> 
> > did they at least gave you a margin of time to unsuspend or something?
> 
> and they said the ban will lift in 1 day
at least its not much time, im still waiting for them to reply me

**Comment by paperc07** [comment]
see we need someone to bot an alt account now, appeal the suspension, get the one day ban, start botting again after the 24hr, get banned again, appeal again, see how long the next one is and just keep doing it on an alt so we can see what they do

**Comment by methreals** [comment]
> see we need someone to bot an alt account now, appeal the suspension, get the one day ban, start botting again after the 24hr, get banned again, appeal again, see how long the next one is and just keep doing it on an alt so we can see what they do

It's 3 days, third ban is 7 days, fourth ban..no one has it yet!

**Comment by nixtynine** [comment]
> > see we need someone to bot an alt account now, appeal the suspension, get the one day ban, start botting again after the 24hr, get banned again, appeal again, see how long the next one is and just keep doing it on an alt so we can see what they do
> 
> It's 3 days, third ban is 7 days, fourth ban..no one has it yet!

4th ban is 10 days, 5th ban - you get all the infinity stones hahah

**Comment by Shops** [comment]
I also just got banned. 1 day suspended for 3rd party software

**Comment by isaiahd1** [comment]
you guys are beyond me. i was scared of 1 ban LOL

**Comment by meteora9479** [comment]
I also got banned. I thought Agatha deck is safe due to officially auto-play, but I'm wrong. LOL

**Comment by paperc07** [comment]
well if anyone is curious I didn't stop..... :( I may have an addiction but still not hit and im finally up to level 45, I was in the rank 29-32 forever, Season pass level is 101 and CL is up to like 3300+

**Comment by paperc07** [comment]
Alright they got me I ended at I think 106 rank 50 and CL 3400

**Comment by Pantasd** [comment]
the only solution is another kind of bot https://graor.com/marvel-snap-bot
